misc/python/materialize/cli/ci_coverage_pr_report.py

```diff
@@ -103,14 +103,17 @@ def mark_covered_lines(
             not unittests_have_run
         ), "Call mark_covered_lines for unit tests last in order to get correct code coverage reports"
 
+    # There will always be an SF line specifying a file before a DA line
+    # according to the lcov tracing file format definition
+    file = None
+
     for line in open(lcov_file):
         line = line.strip()
         if not line:
             continue
         if line == "end_of_record":
             continue
         method, content = tuple(line.strip().split(":", 1))
-        file = "__SENTINEL_VERY_UNLIKELY_TO_MATCH_ANY_FILE_NAME"
```

Comment:
This is wrong because it resets the file after each line. Instead we have to persist the file from the last SF line.